internal/pkg/aws/ec2/ec2.go

```diff
@@ -0,0 +1,105 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package ec2 provides a client to make API requests to Amazon Elastic Compute Cloud.
+package ec2
+
+import (
+	"fmt"
+	"github.com/aws/copilot-cli/internal/pkg/deploy"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+
+const (
+	defaultForAZFilterName = "default-for-az"
+)
+
+// Names for tag filters
+var (
+	TagFilterNameForApp = fmt.Sprintf("tag:%s", deploy.AppTagKey)
+	TagFilterNameForEnv = fmt.Sprintf("tag:%s", deploy.EnvTagKey)
+)
+
+var (
+	// FilterForDefaultVPCSubnets is a pre-defined filter for the default subnets at the availability zone.
+	FilterForDefaultVPCSubnets = Filter {
+		Name:   defaultForAZFilterName,
+		Values: []string{"true"},
+	}
+)
+
+type api interface {
+	DescribeSubnets(*ec2.DescribeSubnetsInput) (*ec2.DescribeSubnetsOutput, error)
+	DescribeSecurityGroups(*ec2.DescribeSecurityGroupsInput) (*ec2.DescribeSecurityGroupsOutput, error)
+}
+
+// Filter contains the name and values of a filter.
+type Filter struct {
```

Comment:
Since this is the input to a function - that's used out of the EC2 context - I'm not sure what I should put as Name and Value. 

Perhaps you can add comments to them

```golang
// Name is the name of the subnet bla bla bla
Name string
// Values are the bla bla bla
Values []string
```